packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -228,135 +272,256 @@ export interface SdkBodyModelPropertyType extends SdkModelPropertyTypeBase {
+export type CollectionFormat = "multi" | "csv" | "ssv" | "tsv" | "pipes";
 
-interface SdkHeaderParameter extends SdkModelPropertyTypeBase {
+export interface SdkHeaderParameter extends SdkModelPropertyTypeBase {
   kind: "header";
   collectionFormat?: CollectionFormat;
   serializedName: string;
 }
 
-interface SdkQueryParameter extends SdkModelPropertyTypeBase {
+export interface SdkQueryParameter extends SdkModelPropertyTypeBase {
   kind: "query";
   collectionFormat?: CollectionFormat;
   serializedName: string;
 }
 
-interface SdkPathParameter extends SdkModelPropertyTypeBase {
+export interface SdkPathParameter extends SdkModelPropertyTypeBase {
   kind: "path";
   urlEncode: boolean;
-  validation: SdkValidation;
   serializedName: string;
   optional: false;
 }
 
-interface SdkBodyParameter extends SdkModelPropertyTypeBase {
+export interface SdkBodyParameter extends SdkModelPropertyTypeBase {
   kind: "body";
   optional: boolean;
   contentTypes: string[];
   defaultContentType: string;
 }
 
-interface SdkOperationParameters {
-  __raw?: HttpOperationParameters;
-  parameters: (SdkQueryParameter | SdkHeaderParameter | SdkPathParameter)[];
-  body?: SdkBodyParameter;
+export type SdkHttpParameter =
+  | SdkQueryParameter
+  | SdkPathParameter
+  | SdkBodyParameter
+  | SdkHeaderParameter;
+
+export interface SdkMethodParameter extends SdkModelPropertyTypeBase {
+  kind: "method";
 }
 
-interface SdkResponseHeader {
-  __raw?: ModelProperty;
+export interface SdkServiceResponseHeader {
+  __raw: ModelProperty;
   serializedName: string;
   type: SdkType;
+  description?: string;
+  details?: string;
 }
 
-interface SdkResponse {
-  __raw?: HttpOperationResponse;
-  statusCodes: HttpStatusCodes[];
-  type: SdkType;
-  doc: string;
-  headers: SdkResponseHeader[];
+export interface SdkMethodResponse {
+  kind: "method";
+  type?: SdkType;
+  responsePath?: string; // how to map service response -> method response (e.g. paging). If undefined, it's a 1:1 mapping
 }
 
-interface SdkOperationBase<TOperation> {
-  __raw?: HttpOperation;
-  path: string;
-  access: AccessFlags;
-  verb: HttpVerb;
-  parameters: SdkOperationParameters;
-  responses: SdkResponse[];
-  exceptions: SdkResponse[];
-  overloads?: TOperation[];
-  overloading?: TOperation;
+export interface SdkServiceResponse {
+  type?: SdkType;
+  headers: SdkServiceResponseHeader[];
+}
+
+export interface SdkHttpResponse extends SdkServiceResponse {
+  __raw: HttpOperationResponse;
+  kind: "http";
+  contentTypes?: string[];
+  defaultContentType?: string;
 }
 
-interface SdkBasicOperation extends SdkOperationBase<SdkBasicOperation> {
+interface SdkServiceOperationBase {}
+
+export type SdkParameter = SdkEndpointParameter | SdkCredentialParameter | SdkMethodParameter;
+
+export interface SdkHttpOperation extends SdkServiceOperationBase {
+  __raw: HttpOperation;
+  kind: "http";
+  path: string;
+  verb: HttpVerb;
+  pathParams: SdkPathParameter[];
```

Comment:
Historically, we've tried to keep the order of the generated params consistent with the order from OpenAPI. With the params split like this, we can no longer achieve this. Is this intentional and if so why is that?

Reply:
that's a good point, we wanted to separate them out specifically for http calls since some languages do different things for each parameter. Originally we had also a `parameters` property that had the same ordering as the spec, I removed in thinking it was a duplication, but you're right this is somethign we should have. Let me add it back

Reply:
Adding `parameters` back sounds great.

Is there value in keeping the others? They duplicate the source. If other languages have a need to split them up then IMO they should do it.

Reply:
It also complicates the API and might not be clear when to use `parameters` or the (duplicated) variants.

The API should try to push consumers into the "pit of success" as much as possible.

Reply:
that is fair, let me remove and then add back if needed later. thanks!

Reply:
I see that header/path/query params have been consolidated into `parameters` which is great. Should `bodyParams` also be consolidated? Is it possible in tsp to model an operation with an ordered set of params that include body/header/path/query?
```
fooOp(id: string, w: Widget, count: int32)
```
Where `id` is a header param, `w` is a body param, and `count` is a query param. I presume we'd want the emitted param order to match the tsp defined order.